static-build/qemu/Dockerfile

```diff
@@ -0,0 +1,46 @@
+from ubuntu:16.04
+
+ARG QEMU_REPO
+# commit/tag/branch
+ARG QEMU_VERSION
+
+WORKDIR /root/qemu
+RUN apt-get update
+RUN apt-get install -y \
+	    autoconf \
+	    automake \
+	    bc \
+	    bison \
+	    cpio \
+	    flex \
+	    gawk \
+	    libaudit-dev \
+	    libcap-dev \
+	    libcap-ng-dev \
+	    libdw-dev \
+	    libelf-dev \
+	    libglib2.0-0 \
+	    libglib2.0-dev \
+	    libglib2.0-dev git \
+	    libltdl-dev \
+	    libpixman-1-dev \
+	    libtool \
+	    pkg-config \
+	    pkg-config \
+	    python \
+	    python-dev \
+	    rsync \
+	    zlib1g-dev
+
+RUN cd  .. && git clone "${QEMU_REPO}" qemu
+RUN git checkout "${QEMU_VERSION}"
+RUN git clone https://github.com/qemu/capstone.git capstone
+RUN git clone  https://github.com/qemu/keycodemapdb.git ui/keycodemapdb
+
+ADD configure-hypervisor.sh /root/configure-hypervisor.sh
+
+RUN /root/configure-hypervisor.sh -s kata-qemu | xargs ./configure --prefix=/opt/kata --with-pkgversion=kata-static
+RUN make clean
+RUN make -j8
```

Comment:
any reason for 8 here?  What if I have less?  What if I have many many more?

Reply:
I think @grahamwhaley was saying that `make -j` can DoS your system so it's a shame there isn't a:

```
make -J # use all except 1 CPUs to avoid killing your machine
```

... but maybe simulate with:

```
cpus=$(expr $(nproc) - 1)
make -j $cpus
```

There is of course an assumption that you've got `> 1` CPUs there though.

Reply:
Heh. I was just thinking, doesn't `-j` do the right thing... but, not really - it launches as many jobs in parallel as it can!
I think elsewhere we have `-j$(nproc)` irrc, but there are the two other options:
- as per @jodh-intel says, `$(nproc)-1` might allow you to still read your email :-)
- `$(nproc)+1`, in theory, means there is always one job ready and waiting to build the instant one of the in-flight jobs completes, so may get you (a marginally) faster build.

I vote for the one in the middle then - `-j$(nproc)` :-)

Reply:
my bad, I hard-coded that number, updated.